utilslib/dr.py

```diff
@@ -203,17 +203,15 @@ def __init__(self, *args, **kwargs):
         """
         super(K8s, self).__init__()
 
-        # Use in cluster config when deployed to cluster
-        config.load_kube_config()
         if 'cluster_name' in kwargs:
-            self.cluster_name = kwargs.get("cluster_name")
+            config.load_kube_config()
```

Comment:
I think this might break a bunch of other tests as i'm using it in other tests

Reply:
all tests are running for me!

`make coverage
venv/bin/python3 -m pytest --cov=utilslib
======================================================================= test session starts ========================================================================
platform linux -- Python 3.6.9, pytest-5.3.5, py-1.8.1, pluggy-0.13.1 -- /home/pcarlton/src/github.com/paulcarlton-ww/k8s-dr-utils/venv/bin/python3
cachedir: .pytest_cache
rootdir: /home/pcarlton/src/github.com/paulcarlton-ww/k8s-dr-utils, inifile: pytest.ini, testpaths: tests
plugins: cov-2.8.1, mock-2.0.0
collected 15 items                                                                                                                                                 

tests/backup_test.py::test_backup 
-------------------------------------------------------------------------- live log setup --------------------------------------------------------------------------
INFO     botocore.credentials:credentials.py:1073 Found credentials in environment variables.
PASSED                                                                                                                                                       [  6%]
tests/k8s_test.py::test_read_namespace PASSED                                                                                                                [ 13%]
tests/k8s_test.py::test_list_kind_cm PASSED                                                                                                                  [ 20%]
tests/k8s_test.py::test_list_kind_deployment PASSED                                                                                                          [ 26%]
tests/k8s_test.py::test_read_kind_cm PASSED                                                                                                                  [ 33%]
tests/k8s_test.py::test_get_cluster_info PASSED                                                                                                              [ 40%]
tests/retrieve_test.py::test_get_bucket_keys_exists PASSED                                                                                                   [ 46%]
tests/retrieve_test.py::test_get_bucket_keys_no_contents PASSED                                                                                              [ 53%]
tests/retrieve_test.py::test_get_bucket_item PASSED                                                                                                          [ 60%]
tests/retrieve_test.py::test_get_s3_namespaces PASSED                                                                                                        [ 66%]
tests/s3_test.py::test_parse_key_correctly PASSED                                                                                                            [ 73%]
tests/s3_test.py::test_parse_key_incorrectley PASSED                                                                                                         [ 80%]
tests/store_test.py::test_store_object PASSED                                                                                                                [ 86%]
tests/store_test.py::test_delete_object_from_versioned_bucket PASSED                                                                                         [ 93%]
tests/store_test.py::test_delete_object_from_unversioned_bucket PASSED                                                                                       [100%]

----------- coverage: platform linux, python 3.6.9-final-0 -----------
Name                   Stmts   Miss  Cover
------------------------------------------
utilslib/__init__.py       0      0   100%
utilslib/dr.py           255     64    75%
utilslib/library.py       84     27    68%
------------------------------------------
TOTAL                    339     91    73%


======================================================================== 15 passed in 5.61s ========================================================================`

Reply:
I was worrying about nothing then